Sources/BraintreeApplePay/BTApplePayClient.swift

```diff
@@ -0,0 +1,129 @@
+import Foundation
+import PassKit
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+/// Used to process Apple Pay payments
+@objcMembers public class BTApplePayClient: NSObject {
+
+    //MARK: - Internal Properties
+
+    /// Exposed for testing to get the instance of BTAPIClient
+    var apiClient: BTAPIClient
+
+    // MARK: - Initializer
+
+    /// Creates an Apple Pay client
+    /// - Parameter apiClient: An API client
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+
+    // MARK: - Public Methods
+
+    @objc(paymentRequest:)
+    public func paymentRequest(completion: @escaping (PKPaymentRequest?, Error?) -> Void) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.apiClient.sendAnalyticsEvent("ios.apple-pay.error.configuration")
+                self.completionHandler(onMainThreadWithPaymentRequest: nil, error: error, completion: completion)
+                return
+            }
+
+            guard let configuration, configuration.isApplePayEnabled else {
+                self.apiClient.sendAnalyticsEvent("ios.apple-pay.error.disabled")
+                self.completionHandler(onMainThreadWithPaymentRequest: nil, error: BTApplePayError.unsupported, completion: completion)
+                return
+            }
+
+            let paymentRequest = PKPaymentRequest()
+            paymentRequest.countryCode = configuration.applePayCountryCode ?? ""
+            paymentRequest.currencyCode = configuration.applePayCurrencyCode ?? ""
+            paymentRequest.merchantIdentifier = configuration.applePayMerchantIdentifier ?? ""
+            paymentRequest.supportedNetworks = configuration.applePaySupportedNetworks ?? []
+
+            self.completionHandler(onMainThreadWithPaymentRequest: paymentRequest, error: nil, completion: completion)
+        }
+    }
+
+    @objc(tokenizeApplePayPayment:completion:)
+    public func tokenize(_ payment: PKPayment, completion: @escaping (BTApplePayCardNonce?, Error?) -> Void) {
```

Comment:
CHANGELOG entry is good enough for me. It may be worth calling out in a migration guide if `tokenize` becomes a common naming convention across modules within the iOS SDK. I'll defer to y'all though.